File: src/main/java/com/fury/game/entity/combat/scripts/BorkCombat.java
package com.fury.game.entity.combat.scripts;

import com.fury.core.model.node.entity.actor.figure.Figure;
import com.fury.core.model.node.entity.actor.figure.mob.Mob;
import com.fury.game.entity.combat.CombatScript;
import com.fury.game.npc.bosses.bork.Bork;
import com.fury.game.system.files.loaders.npc.MobCombatDefinitions;
import com.fury.game.world.update.flag.block.Animation;

public class BorkCombat extends CombatScript {

    @Override
    public Object[] getKeys() {
        return new Object[] { "Bork" };
    }


    @Override
    public int attack(Mob mob, Figure target) {
        final MobCombatDefinitions cdef = mob.getCombatDefinition();
        Bork bork = (Bork) mob;
        if (mob.getHealth().getHitpoints() <= (cdef.getHitpoints() * 0.6) && !bork.isSpawnedMinions()) {
            bork.spawnMinions();
            return 0;
        }
        mob.perform(new Animation(cdef.getAttackAnim()));
        delayHit(mob, 0, target, getMeleeHit(mob, getRandomMaxHit(mob, cdef.getMaxHit(), MobCombatDefinitions.MELEE, target)));
        return mob.getCombat().getAttackSpeed();
    }

}

File: src/main/kotlin/com/fury/game/node/entity/mob/combat/BorkCombat.kt
package com.fury.game.node.entity.mob.combat

import com.fury.core.model.node.entity.actor.figure.Figure
import com.fury.game.entity.character.combat.Hit
import com.fury.game.node.entity.actor.figure.mob.MobCombat
import com.fury.game.npc.bosses.bork.Bork

class BorkCombat(private val mob: Bork): MobCombat(mob) {
    override fun handleHit(hit: Hit) {
        if (hit.weapon != null && hit.weapon.id == 25202)
            mob.isSpawnedMinions = true

        super.handleHit(hit)
    }

    override fun sendDeath(source: Figure?) {
        if (!mob.isSpawnedMinions) {
            mob.health.hitpoints = 1
            return
        }
        mob.controller.killBork()
        if (mob.borkMinion != null) {
            for (n in mob.borkMinion!!) {
                if (n == null || n.isDead)
                    continue
                n.sendDeath(source)
            }
        }

        super.sendDeath(source)
    }
}